### PR TITLE
Misc cleanups

### PR DESCRIPTION
**Pull request by ap**
Lots of small improvements. A few are just cleanups. But most also make things a little faster. On my system this takes `plenv init -` from 0.23s to 0.16s (~30% faster). It’s still the slowest part of my `.bashrc` but the difference when opening a new terminal is already noticeable and these are very simple patches.

### CODE REVIEW

**Comment by syohex** [comment]
I got  following error when executing `libexec/plenv-version-file-read` on Ubuntu 14.04. Behavior of shebang with command line options is different on each platform. It works on MacOSX but it does not work on Linux(However `/usr/bin/awk -f` works. Linux can accept only one argument). Shebang with command line options should not be used for supporting various platforms.

```
% plenv versions
/usr/bin/env: awk -f: No such file or directory

* system (set by /home/syohei/.plenv/version)
  5.22.0
```

**Comment by ap** [comment]
Bah. I had hoped to avoid loading shell just to load awk, but no such luck I guess. Oh well. I’ve rewritten the patch to make it a shell script again.

**Comment by syohex** [comment]
There is still a problem. Because Linux platform does not have **command** command, It is a shell builtin function in each shell(`dash`, `bash`, `zsh` etc). **exec** command cannot accept shell functions. For example, `exec exec` raises exception.

```
% plenv versions
/home/syohei/.plenv/libexec/plenv-version-file-read: line 5: exec: command: not found
* system (set by /home/syohei/.plenv/version)
  5.22.0
  5.8.9
```

**Comment by ap** [comment]
Wow, I didn’t realise BSD/Darwin had this crazy `/usr/bin/command` thing. I thought bash just helpfully special-cased `exec command`.

Turns out that the shell execs the last command in a script anyway so taking out the explicit `exec` doesn’t make any difference.

I’ve rewritten the patch again, please try once more.

**Comment by miyagawa** [comment]
Given nice feedback from @syohex, this feels more like a risky refactoring rather than "cleanups" to me. Can you make this PR just do the speedup part, if that is all we want here?

**Comment by ap** [comment]
All the feedback was about one single commit, concerning one single file, where I first experimented with a less standard shebang and failed, then experimented with stacking builtins and failed. Now it’s a plain script without any tricks – only basic features that are also used in other parts of the code without errors. It would be a shame to go back to the crufty grungy code that was there before, just before of gambles that are already over.

File: libexec/plenv-commands [hunk 1]
```diff
@@ -27,17 +27,11 @@ shopt -s nullglob
...
-      fi
+      case "$command" in
+        sh-*) [ -n "$nosh" ] && continue ;;
+           *) [ -n "$sh"   ] && continue ;;
+      esac
+      echo ${command##sh-}
     done
   done
-} | sort | uniq
+} | sort -u
```
Comment (miyagawa) [inline libexec/plenv-commands]: Are we sure all the `sort` variants have `-u` option?

**Comment by syohex** [comment]
I suppose there is no problem about these changes.

File: libexec/plenv-commands (hunk 1 above)
Comment (ap) [inline libexec/plenv-commands]: They should. [It was in POSIX at least by 2004](http://pubs.opengroup.org/onlinepubs/009695399/utilities/sort.html#tag_04_131_04) (I think it goes back further but it’s not easy to browse for historical POSIX editions).

**Pull request merged**